src/Graphene.SQLServer/SQLReportGenerator.cs

```diff
@@ -111,7 +111,7 @@ public ITrackerReportResults BuildReport(IReportSpecification specification)
                             if (!sqlReader.IsDBNull(timeSlotC))
                             {
                                 measurementDate = new DateTimeOffset(sqlReader.GetDateTime(timeSlotC),
-                                    specification.OffsetTotalsByHours).UtcDateTime;
+                                    new TimeSpan(specification.OffsetTotalsByHours.Hours * -1)).UtcDateTime;
```

Comment:
We need to subtract the offset passed in when shifting the times back from local to UTC.


·         UTC -> Local =Add offset

·         Local -> UTC = Subtract offset



From: Boban Jose [mailto:notifications@github.com]
Sent: Wednesday, January 13, 2016 1:30 PM
To: bobanjose/Graphene
Cc: Owens, Ian
Subject: Re: [Graphene] PS-2972 : Converted returned offset date time values back to UTC for (#36)


In src/Graphene.SQLServer/SQLReportGenerator.cs<https://github.com/bobanjose/Graphene/pull/36#discussion_r49646033>:

> @@ -111,7 +111,7 @@ public ITrackerReportResults BuildReport(IReportSpecification specification)

>                              if (!sqlReader.IsDBNull(timeSlotC))

>                              {

>                                  measurementDate = new DateTimeOffset(sqlReader.GetDateTime(timeSlotC),

> -                                    specification.OffsetTotalsByHours).UtcDateTime;

> +                                    new TimeSpan(specification.OffsetTotalsByHours.Hours * -1)).UtcDateTime;

Can you please clarify why we need the -1 hardcoded there?

—
Reply to this email directly or view it on GitHub<https://github.com/bobanjose/Graphene/pull/36/files#r49646033>.